python/pecos/classical_interpreters/phir_classical_interpreter.py

```diff
@@ -261,8 +261,10 @@ def assign_int(self, cvar, val: int):
             cval &= ~(1 << i)
             cval |= (val & 1) << i
 
-        # mask off bits give the size of the register
-        cval &= (1 << size) - 1
+        if not isinstance(cval, (np.int64, np.int32)):
```

Comment:
I wish I could exclude all signed types instead of enumerating them.

Reply:
@dlucchetti Thoughts on not masking off signed ints?

Reply:
So maybe this is a future selves problem?

Reply:
@dlucchetti @ciaranra It's not clear if there is something actionable here. Are we converging on merging this PR for and do some integration testing?

Like I mentioned elsewhere, I did some of this integration testing using this branch already with the internal emulator and compiler tests. Please let me know if there is more we can do here before merging.

Reply:
It does sound like maybe future problems... although not having a consistent approach gets me a little antsy...

What PR were you considering merging this with?

I haven't looked closely at the tests yet. I was planning to sit down and look at them and play around with the implementation. Should we be moving some tests from the internal emulator or compiler to here for better coverage/regression testing?

Reply:
> What PR were you considering merging this with?

I meant this PR into `development` branch and making a new release. Though it sounds like you don't feel comfortable with it just yet.

> I haven't looked closely at the tests yet. I was planning to sit down and look at them and play around with the implementation.

Thanks, that might be nice. I don't have the best picture of what was happening in the existing `BinArray` with the unsigned bitstring assumptions.

> Should we be moving some tests from the internal emulator or compiler to here for better coverage/regression testing?

That would be great. I don't know if it makes sense to move any compiler tests here, but if the PHIR classical interpreter was also using `BinArray2`, we could create several tests as PHIR input.

Reply:
> I meant this PR into `development` branch and making a new release. Though it sounds like you don't feel comfortable with it just yet.

I think I might be fine with it. Just feels a little bit like weird/undefined behavior. I wonder if the solution would just be to raise an exception if a size argument was given for signed ints when PHIR is translated to the Python representation in PECOS.

> That would be great. I don't know if it makes sense to move any compiler tests here, but if the PHIR classical interpreter was also using `BinArray2`, we could create several tests as PHIR input.

Hmmm, good point... although maybe not worth if we move away from `BinArray2` to the PHIR interpreter...

Alright maybe for this PR I will just look more closely into the tests/play around with it. Although, if there is a obvious place to insert an exception for signed ints with size... that might be good to add.

Reply:
> I wonder if the solution would just be to raise an exception if a size argument was given for signed ints when PHIR is translated to the Python representation in PECOS.

I tried that, but we currently have several test cases that use that, e.g. https://github.com/PECOS-packages/PECOS/blob/development/tests/integration/phir/example1_no_wasm.json#L12-L25

> maybe not worth if we move away from `BinArray2` to the PHIR interpreter...

Oh, is PHIR interpreter going to the primary classical interpreter?

Reply:
> Oh, is PHIR interpreter going to be the primary classical interpreter?

What I mean is that internal emulator does not use PHIR/the PHIR interpreter right now. However, it would be nice to switch it to that so there is one emulation process. The PHIR interpreter doesn't use `BinArray`/`BinArray2`.